src/ikernel.ts

```diff
@@ -759,6 +754,22 @@ interface IKernel extends IDisposable {
   connectToComm(targetName: string, commId?: string): IComm;
 
   /**
+   * Register a comm target handler.
+   *
+   * @param targetName - The name of the comm target.
+   *
+   * @param callback - The callback invoked for a comm open message.
+   *
+   * @returns A disposable used to unregister the comm target.
+   *
+   * #### Notes
+   * Only one comm target can be registered at a time, an existing
+   * callback will be overidden.  A registered comm target handler will take
+   * precedence over a comm which specifies a `target_module`.
+   */
+  registerCommTarget(targetName: string, callback: (comm: IComm, msg: IKernelMessage) => void): IDisposable;
```

Comment:
Can we subclass the IKernelMessage to make an ICommOpenMessage that is passed to the callback, to give some extra type checking information?  Something like this in the ikernel.ts file:

```

/**
 * IOPub comm_open kernel message specification.
 * 
 * See [Comm open](http://jupyter-client.readthedocs.org/en/latest/messaging.html#opening-a-comm).
 */
export
interface IKernelIOPubCommOpenMessage extends IKernelMessage {
  content: {
    comm_id: string;
    target_name: string;
    data: any;
    target_module?: string;
  }
}

/**
 * Check if an IKernelMessage is an iopub comm_open message.
 */
export
function isCommOpenMessage(msg: IKernelMessage): msg is IKernelIOPubCommOpenMessage {
  return msg.header.msg_type === "comm_open";
}
```